File: cadastro-moradores-api/src/main/java/com/elbertribeiro/cadastromoradores/usuario/UsuarioServico.java
package com.elbertribeiro.cadastromoradores.usuario;

import com.elbertribeiro.cadastromoradores.message.MessageResponse;
import org.springframework.stereotype.Service;

import java.util.List;

@Service
public class UsuarioServico {
    final UsuarioRepositorio usuarioRepositorio;

    public UsuarioServico(UsuarioRepositorio usuarioRepositorio) {
        this.usuarioRepositorio = usuarioRepositorio;
    }

    public List<Usuario> buscarListaUsuarios(){
        return usuarioRepositorio.findAll();
    }

    public MessageResponse salvarUsuario(Usuario usuario) {
        return new MessageResponse( "Usuario criado com ID " + usuarioRepositorio.save(usuario).getId());
    }
}

File: cadastro-moradores-api/src/main/kotlin/com/elbertribeiro/cadastromoradores/usuario/UsuarioServico.kt
package com.elbertribeiro.cadastromoradores.usuario

import com.elbertribeiro.cadastromoradores.message.MessageResponseDto
import org.springframework.stereotype.Service

@Service
class UsuarioServico(val usuarioRepositorio: UsuarioRepositorio) {
    fun buscarListaUsuarios(): MutableList<Usuario> {
        return usuarioRepositorio.findAll()
    }

    fun salvarUsuario(usuario: Usuario): MessageResponseDto {
        return createMessageResponse(usuarioRepositorio.save(usuario).id, "Usuario criado com ID")
    }

    private fun createMessageResponse(id: Long, message: String): MessageResponseDto {
        val messageResponseDTO = MessageResponseDto()
        messageResponseDTO.message = "$message $id"
        return messageResponseDTO
    }
}